cli/command/plugin/install.go

```diff
@@ -54,26 +54,14 @@ func newInstallCommand(dockerCli command.Cli) *cobra.Command {
 	return cmd
 }
 
-type pluginRegistryService struct {
-	registry.Service
-}
-
-func (s pluginRegistryService) ResolveRepository(name reference.Named) (*registry.RepositoryInfo, error) {
-	repoInfo, err := s.Service.ResolveRepository(name)
+func resolvePlugnRepository(name reference.Named) (*registry.RepositoryInfo, error) {
+	repoInfo, err := registry.ParseRepositoryInfo(name)
 	if repoInfo != nil {
 		repoInfo.Class = "plugin"
 	}
```

Comment:
We need to check if this is still needed; ISTR this was some snafu on Docker Hub, and may not even be needed anymore. /cc @tianon (ISTR you were on that discussion, but I don't recall the status 🤔)

Reply:
That's correct!  Docker Hub still allows the auth class, but it works fine either way (ie, it is no longer _required_ like it used to be).

Reply:
If that's the case, perhaps we can remove all of this altogether 🤔 🎉 

I tried to be a bit more careful as I'm not even sure this code-path is tested 😬 (plugins are a bit of a corner-case, so it wouldn't surprise me if there isn't a e2e test for this).

Reply:
Gave that a quick go in https://github.com/docker/cli/pull/4114 (still in draft; if that variant would work, I can update this PR 😅)